docker_kernel/kernel.py

```diff
@@ -107,21 +117,55 @@ def do_complete(self, code: str, cursor_pos: int):
         }
 
     def create_build_stage(self, code):
-        """ Add current *_sha1* to the code."""
+        """
+        Add current *_sha1* to the code.
+        Replace base image alias or index with image id.
+        COPY --from=[image alias/index] [src] [dest]
+        -->
+        COPY --from=[image id] [src] [dest]
+        """
+        self.start_a_new_layer(code)
+        code = self.replace_base_image_with_id(code)
         if self._sha1 is not None:
             code = f"FROM {self._sha1}\n{code}"
         return code
 
+    def replace_base_image_with_id(self, code):
+        if '--from=' in code:
+            code_seg = code.split(" ")
+            contain_image = [s for s in code_seg if "--from" in s]
+            image_alias = ''.join(contain_image).split("=")[1]
+            base_image_id = list(self._intermedia_images.values())[int(image_alias)] if image_alias.isdigit() \
+                else self._intermedia_images[image_alias]
+            code = f"{code_seg[0]} --from={base_image_id} {' '.join(code_seg[2:])}"
+        return code
+
+    def start_a_new_layer(self, code):
+        if code.startswith("FROM"):
+            try:
+                alias = code.split('as')[1].strip()
+            except:
+                alias = len(self._intermedia_images)
+            self._current_alias = alias
+            self._intermedia_images[alias] = ''
+
     def build_image(self, code):
         """ Build docker image by passing input to the docker API."""
+
         f = io.BytesIO(code.encode('utf-8'))
-        logs = [] 
-        for logline in self._api.build(fileobj=f, rm=True):
+        logs = []
+        current_working_directory = os.getcwd()
+        dockerfile_path = os.path.join(current_working_directory, 'Dockerfile')
```

Comment:
Don't just use `Dockerfile` as a filename. This might overwrite existing files. Or at least mark this line with a `TODO:...` comment saying we should change this later and this is only for testing the functionality.